pkg/kubectl/cmd/apply_diff_last_applied.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/util/diff"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	applyDiffLastAppliedLong = templates.LongDesc(i18n.T(`
+		Opens up a 2-way diff in the default diff viewer. This should follow the same semantics as git diff.
+		It should accept either a flag --diff-viewer=meld or check the environment variable KUBECTL_EXTERNAL_DIFF=meld.
+		If neither is specified, the diff command should be used.
+		`))
+
+	applyDiffLastAppliedExample = templates.Examples(i18n.T(`
+		# Diff the last-applied-configuration annotations with input file
+		kubectl apply diff-last-applied -f deploy.yaml
+		`))
+)
+
+func NewCmdApplyDiffLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &diff.DiffOptions{Out: out, ErrOut: err}
+	cmd := &cobra.Command{
+		Use:     "diff-last-applied -f FILENAME",
+		Short:   i18n.T("Diff the last-applied-configuration annotation on a live object to match the contents of a file."),
+		Long:    applyDiffLastAppliedLong,
+		Example: applyDiffLastAppliedExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd))
+			cmdutil.CheckErr(options.Validate(f, cmd))
+			cmdutil.CheckErr(options.RunDiffLastApplied(f, cmd))
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.Output, "output", "o", "yaml", "Output format. One of: yaml|json.")
```

Comment:
>at most twice

Even two times would be enough for this to be consolidated to a single helper function. Avoid duplicating code and especially avoid duplicating string literals.

Incidentally, `--output` is exceedingly common and is defined at least 7 times in kubectl.

```
$ rg "Flags" | rg "String" | rg '"output"' | rg -v _test 
1:6:25:cmd/apply_edit_last_applied.go:98:	cmd.Flags().StringVarP(&options.Output, "output", "o", "yaml", "Output format. One of: yaml|json.")
2:7:27:cmd/apply_view_last_applied.go:75:	cmd.Flags().StringP("output", "o", "", "Output format. Must be one of yaml|json")
3:17:97:cmd/edit.go:109:	cmd.Flags().StringVarP(&options.Output, "output", "o", "yaml", "Output format. One of: yaml|json.")
4:55:262:cmd/version.go:59:	cmd.Flags().String("output", "", "output format, options available are yaml and json")
5:144:788:cmd/util/printing.go:56:	cmd.Flags().StringP("output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
6:145:791:cmd/util/printing.go:61:	cmd.Flags().StringVarP(output, "output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
7:146:794:cmd/util/printing.go:66:	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|wide|name|custom-columns=...|custom-columns-file=...|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=... See custom columns [http://kubernetes.io/docs/user-guide/kubectl-overview/#custom-columns], golang template [http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template [http://kubernetes.io/docs/user-guide/jsonpath].")
```